prettyprinter/src/Prettyprinter/Internal.hs

```diff
@@ -292,6 +293,9 @@ class Pretty a where
 --   , (1, [2, 2, 2, 2, 2, 2, 2, 2])
 --   , (1, [2, 2, 2, 2, 2, 2, 2, 2]) ] ]
 
+instance ann ~ Void => Pretty (Doc ann) where
+  pretty = vacuous
```

Comment:
Doesn't `vacuous` perform useless work traversing the `Doc`?! It's implemented as `fmap absurd`.

How about this instead?!

```suggestion
  pretty = unsafeCoerce
```

Reply:
```haskell
pretty = unsafeCoerce
```

You forgot a two hundred essay justifying it. 

E.g. I'm not sure what happens when there are bottoms, `annotate (error "surprise") doc :: Doc Void`.

Reply:
To be clear: I won't author a commit with `unsafeCoerce` even `fmap absurd` is almost always can be replaced by it.